scripts/ci/check_compliance.py

```diff
@@ -262,6 +267,39 @@ def run(self):
         for file in path.glob("**/board.yml"):
             self.check_board_file(file, vendor_prefixes)
 
+
+class ClangFormatCheck(ComplianceTest):
+    """
+    Check if clang-format reports any issues
+    """
+    name = "ClangFormat"
+    doc = "See https://docs.zephyrproject.org/latest/contribute/guidelines.html#clang-format for more details."
+    path_hint = "<git-top>"
+
+    def run(self):
+        for file in get_files():
+            diff = subprocess.Popen(('git', 'diff', '-U0', '--no-color', COMMIT_RANGE, '--', file),
+                                    stdout=subprocess.PIPE,
+                                    cwd=GIT_TOP)
+            try:
+                subprocess.run(('clang-format-diff.py', '-p1'),
+                               check=True,
+                               stdin=diff.stdout,
+                               stdout=subprocess.PIPE,
+                               stderr=subprocess.STDOUT,
+                               cwd=GIT_TOP)
+
+            except subprocess.CalledProcessError as ex:
+                patchset = unidiff.PatchSet.from_string(ex.output, encoding="utf-8")
+                for patch in patchset:
+                    for hunk in patch:
+                        # Set the annotation to the last line of the hunk
+                        line = hunk.source_start
+                        end_line = hunk.source_start + hunk.source_length - 1
+                        self.fmtd_failure("notice", "ClangFormat", file, line=line,
+                                          end_line=end_line, desc=f'\r\n{hunk}')
```

Comment:
Sorry but I still think the notice should appear at the location of the first offending line, especially if you are to make the "diff" appear as part of the message, since you want people to be able to easily match the diff to what's below/after the notice, not above (IMO).

Please see https://github.com/zephyrproject-rtos/zephyr/pull/75165/files also for suggestions on how to make the notice even more useful, namely:

- "ClangFormat" is a bit too succinct of a title, IMO
- Outputting the hunk "header" is unnecessary, and it might make sense to align the diff at the beginning of the annotation

BEFORE:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/1fb539c0-b85f-4b21-8163-d62e4f8a6c96)


AFTER:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/128251/5ec99a85-f836-4343-8fb6-361050ddfff2)

Reply:
I've updated the PR, but simplified the handling to skip the first and last 3 lines, as these were always added surrounding the hunk.

Reply:
Well, I got confused with that new way and I think the BEFORE looks much better, becase the second violation:

![image](https://github.com/zephyrproject-rtos/zephyr/assets/180017/ccf6d4f8-cbdf-4b28-a65e-d6be20b982b3)

 seems completely disconnected from the bot comment